fdbserver/VersionedBTree.actor.cpp

```diff
@@ -8063,6 +8096,10 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 			ASSERT(encryptionKeyProvider->expectedEncodingType() == EncodingType::AESEncryptionV1);
 			encodingType = EncodingType::AESEncryptionV1;
 			m_keyProvider = encryptionKeyProvider;
+		} else if (g_network->isSimulated() && logID.hash() % 2 == 0) {
```

Comment:
I think it's sufficient to rely on standard buggify probabilities here, so just
```
else if (BUGGIFY) {
```

Reply:
BUGGIFY is causing RedwoodBTreeUnexpectedNodeEncoding failure in CI. Reverting for now and will take another look later.

Reply:
It is causing failure because simply doing `BUGGIFY` does not account for restart, and when that happen, encoding type can switch from XOR back to XXHash. Redwood will enforce the XOR encoding type, but it fail to set the correct key provider in this case. Will fix in separate PR.